packages/graphql-transformers-e2e-tests/src/__tests__/ModelConnectionWithKeyTransformer.e2e.test.ts

```diff
@@ -0,0 +1,452 @@
+            projectId
+            name
+        }
+    }
+    `, {});
+
+    const queryResponse = await GRAPHQL_CLIENT.query(`
+    query ListCProjects {
+        listCProjects {
+            items {
+                projectId
+                name
+                team {
+                    teamId
+                    name
+                    project {
+                        projectId
+                        name
+                    }
+                }
+            }
+        }
+    }
+    `, {})
+    expect(queryResponse.data.listCProjects).toBeDefined()
+    const items = queryResponse.data.listCProjects.items
+    expect(items.length).toEqual(1)
+    expect(items[0].projectId).toEqual('P1')
+    expect(items[0].team).toBeDefined()
+    expect(items[0].team.teamId).toEqual('T1')
+    expect(items[0].team.project).toBeDefined()
+    expect(items[0].team.project.projectId).toEqual('P1')
+})
+
+test('Named connection 2 way navigation, with with custom @key fields 1:M', async () => {
+    await GRAPHQL_CLIENT.query(`
+    mutation CreateDProject {
+        createDProject(input: {projectId: "P1", name: "P1"}) {
+            projectId
+            name
+        }
+    }
+    `, {});
+
+    await GRAPHQL_CLIENT.query(`
+    mutation CreateDTeam {
+        createDTeam(input: {teamId: "T1", name: "Team 1", dTeamProjectId: "P1"}) {
+            teamId
+            name
+        }
+    }
+    `, {});
+
+    await GRAPHQL_CLIENT.query(`
+    mutation CreateDTeam {
+        createDTeam(input: {teamId: "T2", name: "Team 2", dTeamProjectId: "P1"}) {
+            teamId
+            name
+        }
+    }
+    `, {});
+
+    const queryResponse = await GRAPHQL_CLIENT.query(`
+    query ListDProjects {
+        listDProjects {
+            items {
+                projectId
+                name
+                teams {
+                    items {
+                        teamId
+                        name
+                        project {
+                            projectId
+                            name
+                        }
+                    }
+                }
+            }
+        }
+    }
+    `, {})
+    expect(queryResponse.data.listDProjects).toBeDefined()
+    const items = queryResponse.data.listDProjects.items
+    expect(items.length).toEqual(1)
+    expect(items[0].projectId).toEqual('P1')
+    expect(items[0].teams).toBeDefined()
+    expect(items[0].teams.items).toBeDefined()
+    expect(items[0].teams.items.length).toEqual(2)
+    expect(items[0].teams.items[0].teamId).toEqual('T1')
+    expect(items[0].teams.items[0].project).toBeDefined()
+    expect(items[0].teams.items[0].project.projectId).toEqual('P1')
+    expect(items[0].teams.items[1].teamId).toEqual('T2')
+    expect(items[0].teams.items[1].project).toBeDefined()
+    expect(items[0].teams.items[1].project.projectId).toEqual('P1')
+})
+
+test('Unnamed connection with sortField parameter only #2100', async () => {
```

Comment:
Can you add listModel test and see if the items are sorted as expected as well

Reply:
That case is already covered by @key tests, what is new here is connection via sort field.